### PR TITLE
Add @projectedName and update service version

### PR DESCRIPTION
**Pull request by joheredi**
This PR adds @friendlyName on object properties when the serialized name is different to the schema name.

Also moves the service version to the @service decorator

### CODE REVIEW

**Review by timotheeguerin — APPROVED** [review]

File: packages/extensions/openapi-to-cadl/test/complexObject/cadl-output/models.cadl [hunk 1]
```diff
@@ -16,12 +16,14 @@ Unique identifier representing a specific product for a given latitude &
 longitude. For example, uberX in San Francisco will have a different product_id
 than uberX in Los Angeles.
 """)
+  @friendlyName("productId")
```
Comment (catalinaperalta) [inline packages/extensions/openapi-to-cadl/test/complexObject/cadl-output/models.cadl:19]: I'm seeing an error locally when this decorator is added over model properties. Asking offline about it.

File: packages/extensions/openapi-to-cadl/test/nh/cadl-output/models.cadl [hunk 2]
```diff
@@ -170,39 +170,78 @@ model NotificationHubEntryModel {
...
   type?: NotificationHubContentModelType;
+
+  @friendlyName("notificationHubDescription")
   NotificationHubDescription?: NotificationHubDescriptionModel;
 }
 
 model NotificationHubDescriptionModel {
+  @friendlyName("xmlnsI")
   "xmlns:i"?: NotificationHubDescriptionModelXmlnsI;
+
```
Comment (catalinaperalta) [inline packages/extensions/openapi-to-cadl/test/nh/cadl-output/models.cadl:181]: Based off of the offline discussion, I think the output here should look like:
```suggestion
  @projectedNames("json", "xmlns:i")
  "xmlnsI"?: NotificationHubDescriptionModelXmlnsI;
 
```
cc @timotheeguerin

Comment (catalinaperalta) [reply packages/extensions/openapi-to-cadl/test/nh/cadl-output/models.cadl:181]: where the serialized name is in projected names

File: packages/extensions/openapi-to-cadl/src/utils/decorators.ts [hunk 3]
```diff
@@ -72,8 +72,8 @@ export function getPropertyDecorators(property: Property): CadlDecorator[] {
 
   if (property.serializedName !== property.language.default.name) {
     decorators.push({
-      name: "friendlyName",
-      arguments: [property.language.default.name],
+      name: "projectedName",
+      arguments: ["json", property.serializedName],
```
Comment (catalinaperalta) [inline packages/extensions/openapi-to-cadl/src/utils/decorators.ts:76]: Should `json` be hardcoded here? Or could there be cases where it's `xml` for instance? I think that info probably shows up somewhere in the tree

Comment (timotheeguerin) [reply packages/extensions/openapi-to-cadl/src/utils/decorators.ts:76]: I think cadl is lacking full xml support for now so I'm fine keeping it like that until we want to convert an xml service and get the bigger picture

**Pull request merged**